### DISCUSSION TITLE
Declaration merging with '@prisma/client'

### DISCUSSION THREAD

**Discussion by Dezzley** [prisma/prisma#2898] [Q&A] [answered]
Greetings, I have plenty of custom-generated types that are used for defining joins or partial selections, for example: 
```javascript
type ReservationWithItem = ReservationGetPayload<{
	include: {
		item: true,
	}
}>
```
I want these types to be available globally by declaring a module in my separate `prisma.client.d.ts` file and merging it with` @prisma/client` module.

What is the best way/practice of doing this?

**Comment by ryands17** [comment, marked as answer]
Hey @Dezzley :wave: 
I think the best way would be to create a separate `index.d.ts` file that is available globally and then add those there. I don't think merging with the `@prisma/client` module directly is possible, so creating a global declaration specific to your project would be the best option.

**Comment by ludralph** [comment]
Hi @Dezzley,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
